drivers/flash/flash_stm32_ospi.c

```diff
@@ -0,0 +1,1499 @@
+		.FlashId            = HAL_OSPI_FLASH_ID_1,
+		.InstructionMode    = ((dev_cfg->data_mode == OSPI_SPI_MODE)
+				? HAL_OSPI_INSTRUCTION_1_LINE
+				: HAL_OSPI_INSTRUCTION_8_LINES),
+		.InstructionDtrMode = ((dev_cfg->data_rate == OSPI_DTR_TRANSFER)
+				? HAL_OSPI_INSTRUCTION_DTR_ENABLE
+				: HAL_OSPI_INSTRUCTION_DTR_DISABLE),
+		.InstructionSize    = ((dev_cfg->data_mode == OSPI_SPI_MODE)
+				? HAL_OSPI_INSTRUCTION_8_BITS
+				: HAL_OSPI_INSTRUCTION_16_BITS),
+		.Instruction        = ((dev_cfg->data_mode == OSPI_SPI_MODE)
+				? JESD216_CMD_READ_SFDP
+				: JESD216_OCMD_READ_SFDP),
+		.Address = addr,
+		.AddressDtrMode     = ((dev_cfg->data_rate == OSPI_DTR_TRANSFER)
+				? HAL_OSPI_ADDRESS_DTR_ENABLE
+				: HAL_OSPI_ADDRESS_DTR_DISABLE),
+		.AddressSize        = ((dev_cfg->data_mode == OSPI_SPI_MODE)
+				?  HAL_OSPI_ADDRESS_24_BITS
+				:  HAL_OSPI_ADDRESS_32_BITS),
+		.AlternateBytesMode = HAL_OSPI_ALTERNATE_BYTES_NONE,
+		.DataMode           = ((dev_cfg->data_mode == OSPI_SPI_MODE)
+				? HAL_OSPI_DATA_1_LINE
+				: HAL_OSPI_DATA_8_LINES),
+		.DataDtrMode        = ((dev_cfg->data_rate == OSPI_DTR_TRANSFER)
+				? HAL_OSPI_DATA_DTR_ENABLE
+				: HAL_OSPI_DATA_DTR_DISABLE),
+		.DummyCycles        = ((dev_cfg->data_mode == OSPI_SPI_MODE) ? 8U : 20U),
+		.DQSMode            = ((dev_cfg->data_rate == OSPI_DTR_TRANSFER)
+				? HAL_OSPI_DQS_ENABLE
+				: HAL_OSPI_DQS_DISABLE),
+		.SIOOMode           = HAL_OSPI_SIOO_INST_EVERY_CMD,
+	};
+
+	cmd.NbData = size;
+
+	HAL_StatusTypeDef hal_ret;
+
+	hal_ret = HAL_OSPI_Command(&dev_data->hospi, &cmd,
+				HAL_OSPI_TIMEOUT_DEFAULT_VALUE);
+	if (hal_ret != HAL_OK) {
+		LOG_ERR("%d: Failed to send OSPI instruction", hal_ret);
+		return -EIO;
+	}
+
+	hal_ret = HAL_OSPI_Receive(&dev_data->hospi, data, HAL_OSPI_TIMEOUT_DEFAULT_VALUE);
+	if (hal_ret != HAL_OK) {
+		LOG_ERR("%d: Failed to read data", hal_ret);
+		return -EIO;
+	}
+
+#endif /* sfdp_bfp */
+	dev_data->cmd_status = 0;
+
+	return 0;
+}
+
+static bool ospi_address_is_valid(const struct device *dev, off_t addr,
+				  size_t size)
+{
+	const struct flash_stm32_ospi_config *dev_cfg = DEV_CFG(dev);
+	size_t flash_size = dev_cfg->flash_size;
+
+	return (addr >= 0) && ((uint64_t)addr + (uint64_t)size <= flash_size);
+}
+
+/*
+ * This function Polls the WIP(Write In Progress) bit to become to 0
+ * in nor_mode SPI/OPI OSPI_SPI_MODE or OSPI_OPI_MODE
+ * and nor_rate transfer STR/DTR OSPI_STR_TRANSFER or OSPI_DTR_TRANSFER
+ */
+static int stm32_ospi_mem_ready(OSPI_HandleTypeDef *hospi, uint8_t nor_mode, uint8_t nor_rate)
+{
+	OSPI_AutoPollingTypeDef s_config = {0};
+	OSPI_RegularCmdTypeDef s_command = ospi_prepare_cmd(nor_mode, nor_rate);
+
+	/* Configure automatic polling mode command to wait for memory ready */
+	s_command.Instruction = (nor_mode == OSPI_SPI_MODE)
+				? SPI_NOR_CMD_RDSR
+				: SPI_NOR_OCMD_RDSR;
+	s_command.AddressMode = (nor_mode == OSPI_SPI_MODE)
+				? HAL_OSPI_ADDRESS_NONE
+				: HAL_OSPI_ADDRESS_8_LINES;
```

Comment:
not exactly, in the ospi_prepare_cmd the the address is set to `HAL_OSPI_ADDRESS_1_LINE` or `HAL_OSPI_ADDRESS_8_LINES`. 
Here in the stm32_ospi_mem_ready(), the AddressMode is set to HAL_OSPI_ADDRESS_NONE. That's the way STM32Cube does. Changing to HAL_OSPI_ADDRESS_1_LINE instead of HAL_OSPI_ADDRESS_NONE works tough
So removing this s_command.AddressMode

Reply:
Right, I read too fast, txs for the update.